PrestaShop/PrestaShop
# Enable 'edit specific price' button on Product page

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | In BackOffice "Catalog" > "Products", currently admins can either "add" or "delete" a "specific price" on the "Edit a Product" page. This PR enables to edit a specific price.
| Type?         | new feature
| Category?     | BO
| BC breaks?    | yes, see below
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/FF-160
| How to test?  | See below



## Possible BC break

Currently, in the "Edit a product" page, **all the forms** are being loaded at once using `PrestaShopBundle\Form\Admin\Product\ProductPrice` Symfony Form.
But the "Add a Specific Price" form is submitted through an AJAX call.

I reused the submit ajax call however, with this PR merged, now the "edit a specific price" button removes the initial "Add a specific price" form and replaces it with another `PrestaShopBundle\Form\Admin\Product\EmbeddedSpecificPrice`. The behavior is iso to `ProductPrice` except is someone overrides it. The submit process has not changed, hooks are preserved.

## How to test

Go on a "Edit a product" page on the BackOffice. Add a specific price. It appears on the page. Now you can edit any of its properties (user group ? customer ? combination ? from and to dates ? how much ? percentage or fixed price ?) and check that the edit process works correctly.

Be careful: in multi-shop context another field appears. Also products with combination have another field that single-combination products do not display.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/9388)
